packages/dynamodb/src/index.ts

```diff
@@ -1,447 +1,398 @@
+      const ExpressionAttributeValues: Record<string, unknown> = {}
+      for (const property in formatedUser) {
+        updateExpression += ` #${property} = :${property},`
+        ExpressionAttributeNames["#" + property] = property
+        ExpressionAttributeValues[":" + property] = formatedUser[property]
+      }
+      updateExpression = updateExpression.slice(0, -1)
+
+      const data = await client.update({
+        TableName,
+        Key: {
+          // next-auth type is incorrect it should be Partial<AdapterUser> & {id: string} instead of just Partial<AdapterUser>
+          pk: `USER#${formatedUser.id as string}`,
+          sk: `USER#${formatedUser.id as string}`,
         },
-        async unlinkAccount(userId, providerId, providerAccountId) {
-          const deleted = await client
-            .delete({
-              TableName,
-              Key: {
-                pk: `USER#${userId}`,
-                sk: `ACCOUNT#${providerId}#${providerAccountId}`,
-              },
-            })
-            .promise()
-
-          return deleted
+        UpdateExpression: updateExpression,
+        ExpressionAttributeNames: ExpressionAttributeNames,
+        ExpressionAttributeValues: ExpressionAttributeValues,
+        ReturnValues: "ALL_NEW",
+      })
+      const attributesDynamo = data.Attributes as Dynamo<AdapterUser>
+      return format.from<AdapterUser>(attributesDynamo)
+    },
+    async deleteUser(userId) {
+      // query all the items related to the user to delete
+      const resQuery = await client.query({
+        TableName,
+        KeyConditionExpression: "#pk = :pk",
+        ExpressionAttributeNames: { "#pk": "pk" },
+        ExpressionAttributeValues: { ":pk": `USER#${userId}` },
+      })
+      if (!resQuery.Items) {
+        return null
+      }
+      const items = resQuery.Items
+      // find the user we want to delete to return at the end of the function call
+      const user = items.find((item) => item.type === "USER")
+      const itemsToDelete = items.map((item) => {
+        return {
+          DeleteRequest: {
+            Key: {
+              sk: item.sk,
+              pk: item.pk,
+            },
+          },
+        }
+      })
+      // batch write commands cannot handle more than 25 requests at once
+      const itemsToDeleteMax = itemsToDelete.slice(0, 25)
+      const param: BatchWriteCommandInput = {
+        RequestItems: { [TableName]: itemsToDeleteMax },
+      }
+      await client.batchWrite(param)
+      return user ? format.from<AdapterUser>(user) : null
+    },
+    async linkAccount(data) {
+      const item = {
+        ...data,
+        pk: `USER#${data.userId}`,
+        sk: `ACCOUNT#${data.provider}#${data.providerAccountId}`,
+        GSI1PK: `ACCOUNT#${data.provider}`,
+        GSI1SK: `ACCOUNT#${data.providerAccountId}`,
+      }
+      await client.put({ TableName, Item: format.to(item) })
+      return item
```

Comment:
```suggestion
      return data
```

Shouldn't return those extra values to the core, it might expose or give hints about what database is being used if account is accessed incorrectly.

Maybe the `format` method could be utilized to returned valued to filter out these unnecessary fields,